core/src/test/java/io/methvin/watchservice/DirectoryWatcherOnDiskTest.java

```diff
@@ -108,7 +113,37 @@ public void moveSubDirectory() throws IOException, ExecutionException, Interrupt
         FileUtils.deleteDirectory(newParent.toFile());
       }
     }
+  }
+
+  @Test
+  public void emitCreateEventWhenFileLocked() throws IOException, ExecutionException, InterruptedException {
+    final CompletableFuture future = this.watcher.watchAsync();
+    final Path child = Files.createTempFile(tmpDir, "child-", ".dat");
+    FileChannel channel = null;
+    FileLock lock = null;
+    try {
+
+      File file = child.toFile();
+      channel = new RandomAccessFile(file, "rw").getChannel();
+      lock = channel.lock();
+
+      try {
+        future.get(5, TimeUnit.SECONDS);
+      } catch (TimeoutException e) {
+        // Expected exception.
```

Comment:
You might just want to throw this since the test would probably fail otherwise, right?

Reply:
It's the same as with other tests, it is expected to timeout since the watcher is not closed.